crates/jstz_proto/src/executor/smart_function.rs

```diff
@@ -444,6 +444,215 @@ pub mod run {
     }
 }
 
+pub mod jstz_run {
+    use tezos_crypto_rs::hash::ContractKt1Hash;
+
+    use super::*;
+    use crate::{executor::withdraw::Withdrawal, operation, receipt};
+
+    const WITHDRAW_PATH: &str = "/withdraw";
+
+    fn validate_withdraw_request(run: operation::RunFunction) -> Result<Withdrawal> {
+        let method = run
+            .method
+            .as_str()
+            .parse::<http::Method>()
+            .map_err(|_| Error::InvalidHttpRequestMethod)?;
+
+        if method != http::Method::POST {
+            return Err(Error::InvalidHttpRequestMethod);
+        }
+
+        let body = match run.body {
+            Some(body) => body,
+            None => Err(Error::InvalidHttpRequestBody)?,
+        };
+
+        let withdrawal: Withdrawal = serde_json::from_str(
+            String::from_utf8(body)
+                .map_err(|_| Error::InvalidHttpRequestBody)?
+                .as_str(),
+        )
+        .map_err(|_| Error::InvalidHttpRequestBody)?;
+
+        Ok(withdrawal)
+    }
+
+    pub fn execute(
+        hrt: &mut impl HostRuntime,
+        tx: &mut Transaction,
+        source: &Address,
+        run: operation::RunFunction,
+        ticketer: &ContractKt1Hash,
+    ) -> Result<receipt::RunFunction> {
+        if run.uri.host() != Some("jstz") {
+            return Err(Error::InvalidHost);
+        }
+        match run.uri.path() {
+            WITHDRAW_PATH => {
+                // TODO: https://linear.app/tezos/issue/JSTZ-77/check-gas-limit-when-performing-native-withdraws
```

Comment:
can I please get access to the linear workspace?

Reply:
I'm not an admin. @johnyob could you please add @alanmarkoTrilitech ?

Reply:
Invited